cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -662,7 +658,8 @@ private List<ApplicationId> decodeAndValidateBatchApplicationRecord(NamespaceId
           throw new BadRequestException("Missing 'name' in the request element for app-id.");
         }
         if (element.getAppVersion() == null) {
-          appIds.add(validateApplicationId(namespaceId, element.getName()));
+          validateApplicationId(namespaceId, element.getName());
+          appIds.add(namespaceId.app(element.getName()));
         } else {
           appIds.add(validateApplicationVersionId(namespaceId, element.getName(), element.getAppVersion()));
```

Comment:
Why do we need both versioned and versionless upgrades?

Reply:
If it's versioned, do we validate it's latest one?

Reply:
same comment as above: https://cdap.atlassian.net/browse/CDAP-19782